### PR TITLE
[nrfconnect] Generate and parse OTA image content header

### PR DESCRIPTION
**Pull request by Damian-Nordic**
Add Nordic-specific OTA image header which encodes contents of the image (app core image, net core image or MCUBoot image).

### CODE REVIEW

File: config/nrfconnect/chip-module/CMakeLists.txt [hunk 1]
```diff
@@ -286,11 +287,32 @@ add_dependencies(chip chip-gn)
...
+        FILE_TYPES ${CHIP_IMAGE_TYPES}
+        FILE_PATHS ${CHIP_IMAGE_PATHS}
+        OUTPUT_FILE ${PROJECT_BINARY_DIR}/${CONFIG_CHIP_OTA_IMAGE_FILE_NAME}.content
+    )
+
     chip_ota_image(chip-ota-image
-        INPUT_FILES ${PROJECT_BINARY_DIR}/app_update.bin
-        OUTPUT_FILE ${PROJECT_BINARY_DIR}/matter_ota.bin
+        INPUT_FILES ${PROJECT_BINARY_DIR}/${CONFIG_CHIP_OTA_IMAGE_FILE_NAME}.content ${CHIP_IMAGE_PATHS}
+        OUTPUT_FILE ${PROJECT_BINARY_DIR}/${CONFIG_CHIP_OTA_IMAGE_FILE_NAME}
```
[the remark this answers is not in the document]
Comment (Damian-Nordic) [reply config/nrfconnect/chip-module/CMakeLists.txt:311]: Good point. Will add.

File: config/zephyr/Kconfig [hunk 2]
```diff
@@ -185,14 +185,30 @@ config APP_LINK_WITH_CHIP
...
-	depends on CHIP_OTA_IMAGE_BUILD
 	help
-	  This option allows one to pass optional arguments to the
-	  ota_image_tool.py script, used for building OTA image.
+	  This option allows one to pass optional arguments to the ota_image_tool.py
+	  script, used for building OTA image.
+
+config CHIP_OTA_IMAGE_INCLUDE_MCUBOOT
+	bool "Include MCUBoot in OTA image"
+	depends on BOOTLOADER_MCUBOOT
```
[the remark this answers is not in the document]
Comment (Damian-Nordic) [reply config/zephyr/Kconfig:206]: I don't think there is. Also, a customer may support a custom bootloader upgrade mechanism so I think we shouldn't make it too strict.

[the remark this answers is not in the document]
Comment (kkasperczyk-no) [reply config/zephyr/Kconfig:206]: Ok, I didn't find it either.

File: config/nrfconnect/chip-module/CMakeLists.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (Damian-Nordic) [reply config/nrfconnect/chip-module/CMakeLists.txt:311]: Done

**Review by kkasperczyk-no — APPROVED** [review]

**Pull request closed without merging**